src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -2343,11 +2469,11 @@ private void clientHandshakeOktaSessionToken(final String db, final OpenIDAuthen
 						final OAuthToken token = blockingGet(grant.getToken(), timeoutSeconds, TimeUnit.SECONDS);
 
 						// Now send the tokens to the database for validation
-						clientHandshakeGCM(token.getTokenHint(), token.getToken(), db, shouldRequestVersion, HandshakeType.OKTA_SESSION_TOKEN);
-
-						LOGGER.log(Level.INFO, "Okta Session Token Exchange complete");
-
-						return;
+						handshakeOktaSessionCompletionStage = clientHandshakeGCM(token.getTokenHint(), token.getToken(), db, shouldRequestVersion, HandshakeType.OKTA_SESSION_TOKEN)
+								.thenApply(successOrEx -> { 
+									LOGGER.log(Level.INFO, () -> "Okta Session Token Exchange complete"); 
+									return successOrEx;
+								});
```

Comment:
This changeset is going to be beautiful, so much delete boilerplate! 😄 
```suggestion
						return clientHandshakeGCM(token.getTokenHint(), token.getToken(), db, shouldRequestVersion, HandshakeType.OKTA_SESSION_TOKEN);
```